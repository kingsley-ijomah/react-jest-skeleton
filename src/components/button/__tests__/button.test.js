import Button from '../button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Button', () => {
  beforeEach(() => {
    render(<Button label="Click me now!" />);
  });

  it('should return a button', () => {
    expect(screen.getByTestId('button-toggle')).toBeTruthy();
  });

  it('should have an attribute of type', () => {
    expect(screen.getByTestId('button-toggle')).toHaveAttribute(
      'type',
      'button'
    );
  });

  it('should have a classname', () => {
    expect(screen.getByTestId('button-toggle')).toHaveClass('button-style');
  });

  it('has the correct label', () => {
    expect(screen.getByTestId('button-toggle')).toHaveTextContent(
      'Click me now!'
    );
  });

  it('has `click me` on page by default', () => {
    expect(screen.getByTestId('click-me')).toBeTruthy();
  });

  it('does not have `you clicked me` on page by default', () => {
    expect(screen.queryByTestId('you-clicked-me')).toBeFalsy();
  });

  it('has `you clicked me` when clicked', () => {
    fireEvent.click(screen.getByTestId('button-toggle'));
    expect(screen.getByTestId('you-clicked-me')).toBeTruthy();
    expect(screen.queryByTestId('click-me')).toBeFalsy();
  });

  it('takes a snapshot', () => {
    const { asFragment } = render(<Button label="Click me now!" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
