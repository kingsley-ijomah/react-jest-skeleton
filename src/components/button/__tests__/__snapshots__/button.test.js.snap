// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button takes a snapshot 1`] = `
<DocumentFragment>
  <button
    class="button-style"
    data-testid="button-toggle"
    type="button"
  >
    Click me now!
  </button>
  <p
    data-testid="click-me"
  >
    Click me!
  </p>
</DocumentFragment>
`;
